pkg/addons/fluxhelmrelease/cni.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package fluxhelmrelease
+
+import (
+	_ "embed" // embedding as []byte does not import the package.
+	"fmt"
+
+	fluxhelmv2beta1 "github.com/fluxcd/helm-controller/api/v2beta1"
+	"github.com/fluxcd/pkg/apis/meta"
+	fluxsourcev1beta2 "github.com/fluxcd/source-controller/api/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// CNIForCluster returns a complete set of Cluster API objects to describe a CNI Configuration
+// installable via Flux resources.
+func CNIForCluster(cluster *clusterv1.Cluster) ([]unstructured.Unstructured, error) {
+	vals, _ := yaml.YAMLToJSON([]byte(`
```

Comment:
This fails for me
```
W0207 22:49:22.504642      13 warnings.go:70] would violate PodSecurity "restricted:latest": host namespaces (hostNetwork=true), allowPrivilegeEscalation != false (container "tigera-operator" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "tigera-operator" must set securityContext.capabilities.drop=["ALL"]), restricted volume types (volume "var-lib-calico" uses restricted volume type "hostPath"), runAsNonRoot != true (pod or container "tigera-operator" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "tigera-operator" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
{"level":"info","ts":"2023-02-07T22:49:34.732Z","msg":"reconcilation finished in 18.590851635s","controller":"helmrelease","controllerGroup":"helm.toolkit.fluxcd.io","controllerKind":"HelmRelease","HelmRelease":{"name":"capi-quickstart-tigera-operator","namespace":"default"},"namespace":"default","name":"capi-quickstart-tigera-operator","reconcileID":"05263640-7a93-4ced-9374-0bdf31d166b1"}
```

But I also don't see any way to configure the chart https://github.com/projectcalico/calico/blob/master/charts/tigera-operator/templates/tigera-operator/02-tigera-operator.yaml

Reply:
Could you try it again? I added some code yesterday (perhaps after you tried it out) to add the necessary labels to the remote namespace and it works for me following the README with a clean env to start from.

Reply:
Oh yeah I was too impatient, thats a warning which is what we expect with the labels.